docs/api-guide/routers.md

```diff
@@ -81,81 +81,64 @@ Router URL patterns can also be namespaces.
 
 If using namespacing with hyperlinked serializers you'll also need to ensure that any `view_name` parameters on the serializers correctly reflect the namespace. In the example above you'd need to include a parameter such as `view_name='api:user-detail'` for serializer fields hyperlinked to the user detail view.
 
-### Extra link and actions
+### Routing for extra actions
 
-Any methods on the viewset decorated with `@detail_route` or `@list_route` will also be routed.
-For example, given a method like this on the `UserViewSet` class:
+Any method on the viewset decorated with `@action` will be included in the generated routes. For example, given a method like this on the `UserViewSet` class:
 
     from myapp.permissions import IsAdminOrIsSelf
-    from rest_framework.decorators import detail_route
+    from rest_framework.decorators import action
 
     class UserViewSet(ModelViewSet):
         ...
 
-        @detail_route(methods=['post'], permission_classes=[IsAdminOrIsSelf])
+        @action(methods=['post'], detail=True, permission_classes=[IsAdminOrIsSelf])
```

Comment:
At this point the `detail` parameter is mysterious.

Reply:
Would it be sufficient to link to the viewset docs first? e.g., "go read this first, then come back here"

Reply:
Yes. (I was thinking along those lines... 🙂)

Perhaps moving the note from ln124 up and into a `Note:` section may be worthwhile. (Who reads about Routers before ViewSets?)

Reply:
Updated the docs. Integrated line 124 into the start of the section.